spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala

```diff
@@ -750,9 +768,19 @@ object TypeConverter {
 
     def targetTypeTag = implicitly[TypeTag[AnyRef]]
 
+    def cassandraOptionToAnyRef(cassandraOption: CassandraOption[_]) = {
```

Comment:
Hmm... I'm still not convinced for this. This looks like messing different levels of encapsulation:
```scala
cassandraOption match {
  case CassandraOption.Value(x) => CassandraOption.Value(nestedConverter.convert(x).asInstanceOf[AnyRef])
  case x: CassandraOption => x
}
```
and make the writer take care of how it is handled.

Disregard this comment if you think it is a load of nonsense :)